plugins/inputs/dns_query/dns_query.go

```diff
@@ -157,6 +136,11 @@ func (d *DnsQuery) parseRecordType() (uint16, error) {
 
 func init() {
 	inputs.Add("dns_query", func() telegraf.Input {
-		return &DnsQuery{}
+		return &DnsQuery{
```

Comment:
you could create a function called `NewDnsQuery` which returns this struct. Then you can call NewDnsQuery() from within init() _and_ from the unit tests.